### ISSUE TITLE
Missing ability to refresh token

### ISSUE THREAD

**Issue by paul** [heroku/platform-api#37]
I don't see a way to refresh the OAuth token once it has expired. The `#connect_oauth` function only accepts the original token, which appears to expire after a few hours. The standard OAuth handshake, and credentials provided by `omniauth-heroku` include a `refresh_token` attribute which may be used to automatically obtain a new token one it has expired. It would be nice if this gem accepted the `refresh_token` as a param (or if it already does, document it somehow. I don't see anything in the gem docs here or in heroic.), and could automatically do the refreshed token exchange.

**Comment by geemus** [comment]
@paul good point. Do you imagine then that it would catch the request error when it expired, get new token and retry? Would you be up for helping toward a PR? Thanks!

**Comment by thebadmonkeydev** [comment]
I'm very interested in this feature...I'll have to implement that basic approach you described for my oauth access any way, might as well push it upstream so I'll start working on a PR as soon as I can.

**Comment by geemus** [comment]
@thebadmonkeydev unfortunately, no progress that I'm aware of.

**Comment by thebadmonkeydev** [comment]
The more I look at this the more I feel like it belongs in heroics and not specifically in platform-api.  @geemus if you agree I'll open the issue there and start working on a solution.

**Comment by thebadmonkeydev** [comment]
The more I look at this the more I feel like it belongs in heroics and not specifically in platform-api.  @geemus if you agree I'll open the issue there and start working on a solution.
